src/main/java/org/datacleaner/extension/elasticsearch/ElasticSearchFullSearchTransformer.java

```diff
@@ -136,22 +156,66 @@ public OutputColumns getOutputColumns() {
 
             final SearchRequestBuilder searchRequestBuilder = new SearchRequestBuilder(client)
                     .setIndices(elasticsearchDatastore.getIndexName()).setTypes(documentType).setQuery(query)
-                    .setSize(1).setSearchType(SearchType.QUERY_AND_FETCH).setExplain(true);
+                    .setSize(getNumberOfResults).setSearchType(SearchType.QUERY_AND_FETCH).setExplain(true);
 
             final SearchResponse searchResponse = searchRequestBuilder.execute().actionGet();
             final SearchHits hits = searchResponse.getHits();
-            if (hits.getTotalHits() == 0) {
+            int totalHits = (int) hits.getTotalHits();
+            if (totalHits == 0) {
+                if (allRowsCollector != null) {
+                    allRowsCollector.putValues( ArrayUtils.add(result, input));
+                }
```

Comment:
Sorry to only see this now, but (and I might have misread this code) this looks like the output data stream will get an output, even if there were zero hits? That seems wrong to me.

It's needed in the "normal" transformer output, otherwise we'd be deleting the row, but it would be unexpected to get a row in an output data stream when there are no results.